compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -607,8 +603,13 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
       case tp if tp.isRef(defn.UnitClass) =>
         Typ(ConstantType(Constant(())), true) :: Nil
       case tp if tp.classSymbol.isAllOf(JavaEnumTrait) =>
+        tp.classSymbol.linkedClass.info.decls.toList.map(_.ensureCompleted())
+        val children = tp.classSymbol.children
```

Comment:
Calls to `.children` could happen in other places, for example from tasty-reflect, can we move the use of `ensureCompleted` inside the definition of `children` itself?

Reply:
Also this can probably be done under an `if (tp.classSymbol.isJavaDefined)`

Reply:
oh, and you can do `decls.foreach` instead of `decls.toList.map`